### PR TITLE
Add Support for Google Authentication

### LINKED ISSUE #17: Add Google Authentication
Using django all-auth library to use Google OAuth Services

### PR DESCRIPTION
**Pull request by Aviral14**
1.Add required settings for all-auth
2.all -auth templates added
3.Delete Unnecessary Files
4.Add Support to make front page of website as login page
5.Add Migrations
6.Remove SignUp functionality from templates
7.Fix problem of each user having a blank username
8.Update Logout Redirect URL
9.Update URL Paths
Close #17

### CODE REVIEW

**Comment by iamkroot** [comment]
If the code is commented, no need to commit it. Don't stage those changes into the commit as they aren't relevant now.

**Comment by Aviral14** [comment]
@iamkroot  There is one more issue.If the user after successfully logging in, uses the back button on the browser to get to the google login page and then logins through some other account again,the "Social Network Login Failure" appears again.I think this is because there is already a logged in user but a request is sent again for logging again through different account.I tried fixing this by implementing -that a new request for login can be sent only if sociallogin.user.is_anonymous is True,in the pre_social_login function .But i think we are redirected to the error page before the request can reach the pre_social_login function. So this isn't working.Can u suggest me some way resolve that?

File: CCIT_Portal/urls.py [hunk 1]
```diff
@@ -15,12 +15,18 @@
...
 
+# from users.views import CustomLoginView
+
+
 urlpatterns = [
     path("admin/", admin.site.urls),
     path("contact/", pages_views.contact, name="contact-us"),
     path("complaints/", include("complaints.urls")),
     path("home/", pages_views.home, name="home"),
+    path("", LoginView.as_view(), name="account_login"),
```
Comment (iamkroot) [inline CCIT_Portal/urls.py]: We shouldn't have the base URL as a login view directly. Keep it at `/login` and the `/home` view should be at `/`, where it redirects to `/login` if not logged in.

Comment (Aviral14) [reply CCIT_Portal/urls.py]: ok i will fix that

**Comment by Aviral14** [comment]
@iamkroot  Fixed the issue of logging again without signing out.Turned out there is an authentication_error adapter which is used before the pre_social_login adapter.This wasn't mentioned in the allauth docs.Had to search through the allauth files manually.Please test the social login now and inform me if it needs any modifications

**Comment by iamkroot** [comment]
> Add #17

Github only recognizes `Fix` and `Close` for issues. In this case, change it to `Close #17`.

**Comment by Aviral14** [comment]
@iamkroot Please review now.

File: users/adapters.py [hunk 2]
```diff
@@ -13,9 +13,23 @@ class CustomSocialAccountAdapter(DefaultSocialAccountAdapter):
...
+            raise ImmediateHttpResponse(redirect("account_login"))
+
+    def authentication_error(
+        self, request, provider_id, error=None, exception=None, extra_context=None
+    ):
+        if not request.user.is_anonymous:
+            messages.error(
+                request,
+                (
+                    f"You are already logged in as {request.user.username}",
```
Comment (iamkroot) [inline users/adapters.py]: Remove the trailing comma here. It is being parsed as a tuple instead of a string.

**Comment by iamkroot** [comment]
Please fix the commit message body. It should not mention every small thing that was done while the PR was in progress. It should elaborate on the major change achieved in the PR, if necessary.

Also, the commit title should have `Add` instead of `Fix` because that is what this commit does.
Here's what the commit message should look like:
```
Add Google Authentication

Implemented using all_auth. Limited to users of BITS email IDs.

Closes #17 
```

**Comment by Aviral14** [comment]
@iamkroot Please Review .

File: users/adapters.py (hunk 2 above)
Comment (iamkroot) [reply users/adapters.py]: @Aviral14 fix this too please

Comment (Aviral14) [reply users/adapters.py]: @iamkroot Fixed!

Comment (iamkroot) [reply users/adapters.py]: The comma is still showing up. Please check.

Comment (Aviral14) [reply users/adapters.py]: check now

**Pull request merged**